test_runner/src/main/kotlin/ftl/args/AndroidArgs.kt

```diff
@@ -83,18 +85,30 @@ class AndroidArgs(
         }
 
         devices.forEach { device -> assertDeviceSupported(device) }
-
-        val validTestMethods = DexParser.findTestMethods(testLocalApk).map { "class ${it.testName}" }
-        validateTestMethods(testTargets, validTestMethods, "Test APK")
+        val filteredTests = getTestMethods(testLocalApk)
 
         testShardChunks = calculateShards(
             testTargets,
-            validTestMethods,
+            filteredTests,
             testTargetsAlwaysRun,
             testShards
         )
     }
 
+    private fun getTestMethods(testLocalApk: String): List<String> {
```

Comment:
Do you really want to fail it with an error? Shouldn't a warning be enough?

Reply:
I thought about warning. For example in the yaml config, if you have an incorrect param that'll produce a warning via the doctor command.

The motivation behind using Flank is for test sharding. Test sharding requires accurate information about test methods. If we're not able to get the test methods, then we can't shard the tests. That seems like a fatal error to me where we'd want to exit immediately and alert the user that something is wrong.

If it was only a warning then the user ends up getting billed for tests that may not have executed as intended.